File: dataModule/src/main/java/com/mobnetic/coinguardian/model/market/LocalBitcoins.java
package com.mobnetic.coinguardian.model.market;

import java.util.List;

import org.json.JSONArray;
import org.json.JSONObject;

import com.mobnetic.coinguardian.model.CheckerInfo;
import com.mobnetic.coinguardian.model.CurrencyPairInfo;
import com.mobnetic.coinguardian.model.Market;
import com.mobnetic.coinguardian.model.Ticker;
import com.mobnetic.coinguardian.model.currency.VirtualCurrency;

public class LocalBitcoins extends Market {

	private final static String NAME = "LocalBitcoins";
	private final static String TTS_NAME = "Local Bitcoins";
	private final static String URL = "https://localbitcoins.com/bitcoinaverage/ticker-all-currencies/";
	private final static String URL_CURRENCY_PAIRS = URL;
	
	public LocalBitcoins() {
		super(NAME, TTS_NAME, null);
	}

	@Override
	public String getUrl(int requestId, CheckerInfo checkerInfo) {
		return URL;
	}
	
	@Override
	protected void parseTickerFromJsonObject(int requestId, JSONObject jsonObject, Ticker ticker, CheckerInfo checkerInfo) throws Exception {
		final JSONObject pairJsonObject = jsonObject.getJSONObject(checkerInfo.getCurrencyPairId());
		ticker.vol = pairJsonObject.getDouble("volume_btc");
		final JSONObject ratesJsonObject = pairJsonObject.getJSONObject("rates");
		ticker.last = ratesJsonObject.getDouble("last");
	}
	
	// ====================
	// Get currency pairs
	// ====================
	@Override
	public String getCurrencyPairsUrl(int requestId) {
		return URL_CURRENCY_PAIRS;
	}
	
	@Override
	protected void parseCurrencyPairsFromJsonObject(int requestId, JSONObject jsonObject, List<CurrencyPairInfo> pairs) throws Exception {
		final JSONArray pairsJsonArray = jsonObject.names();
		for(int i=0; i<pairsJsonArray.length(); ++i) {
			final String currencyCounter = pairsJsonArray.getString(i);
			if(currencyCounter != null) {
				pairs.add(new CurrencyPairInfo(
						VirtualCurrency.BTC,
						currencyCounter,
						currencyCounter));
			}
		}
	}
}

File: dataModule/src/main/java/com/aneonex/bitcoinchecker/datamodule/model/market/LocalBitcoins.kt
package com.aneonex.bitcoinchecker.datamodule.model.market

import com.aneonex.bitcoinchecker.datamodule.model.CheckerInfo
import com.aneonex.bitcoinchecker.datamodule.model.CurrencyPairInfo
import com.aneonex.bitcoinchecker.datamodule.model.Market
import com.aneonex.bitcoinchecker.datamodule.model.Ticker
import com.aneonex.bitcoinchecker.datamodule.model.currency.VirtualCurrency
import org.json.JSONObject

class LocalBitcoins : Market(NAME, TTS_NAME, null) {
    override fun getUrl(requestId: Int, checkerInfo: CheckerInfo): String {
        return URL
    }

    @Throws(Exception::class)
    override fun parseTickerFromJsonObject(requestId: Int, jsonObject: JSONObject, ticker: Ticker, checkerInfo: CheckerInfo) {
        val pairJsonObject = jsonObject.getJSONObject(checkerInfo.currencyPairId!!)
        ticker.vol = pairJsonObject.getDouble("volume_btc")
        val ratesJsonObject = pairJsonObject.getJSONObject("rates")
        ticker.last = ratesJsonObject.getDouble("last")
    }

    // ====================
    // Get currency pairs
    // ====================
    override fun getCurrencyPairsUrl(requestId: Int): String {
        return URL_CURRENCY_PAIRS
    }

    @Throws(Exception::class)
    override fun parseCurrencyPairsFromJsonObject(requestId: Int, jsonObject: JSONObject, pairs: MutableList<CurrencyPairInfo>) {
        val pairsJsonArray = jsonObject.names()!!
        for (i in 0 until pairsJsonArray.length()) {
            val currencyCounter = pairsJsonArray.getString(i)
            if (currencyCounter != null) {
                pairs.add(CurrencyPairInfo(
                        VirtualCurrency.BTC,
                        currencyCounter,
                        currencyCounter))
            }
        }
    }

    companion object {
        private const val NAME = "LocalBitcoins"
        private const val TTS_NAME = "Local Bitcoins"
        private const val URL = "https://localbitcoins.com/bitcoinaverage/ticker-all-currencies/"
        private const val URL_CURRENCY_PAIRS = URL
    }
}